dashboard/app/models/script.rb

```diff
@@ -158,6 +158,17 @@ def hide_pilot_units
 
   validates :published_state, acceptance: {accept: SharedCourseConstants::PUBLISHED_STATE.to_h.values.push(nil), message: 'must be nil, in_development, pilot, beta, preview or stable'}
 
+  after_save :check_course_type_settings
+
+  def check_course_type_settings
+    if is_course?
+      raise 'Published state must be set on the unit if its not in a course.' if published_state.nil?
```

Comment:
nit: right now, scripts can not be in a unit group and not have is_course? set to true. This error message would be clearer (to me) to say "Published state must be set on the unit if it's a standalone unit"